matrix/Matrix.hpp

```diff
@@ -580,6 +580,128 @@ bool isEqual(const Matrix<Type, M, N> &x,
     return true;
 }
 
+namespace typeFunction
+{
+template<typename Type>
+Type min(const Type x, const Type y) {
+    bool x_is_nan = isnan(x);
+    bool y_is_nan = isnan(y);
+    // z > nan for z != nan is required by C the standard
+    if(x_is_nan || y_is_nan) {
+        if(x_is_nan && !y_is_nan) return y;
+        if(!x_is_nan && y_is_nan) return x;
+        return x;
+    }
+    return (x < y) ? x : y;
+}
+template<typename Type>
+Type max(const Type x, const Type y) {
+    bool x_is_nan = isnan(x);
+    bool y_is_nan = isnan(y);
+    // z > nan for z != nan is required by C the standard
+    if(x_is_nan || y_is_nan) {
+        if(x_is_nan && !y_is_nan) return y;
+        if(!x_is_nan && y_is_nan) return x;
```

Comment:
The second condition does the same as default hence it could be left away e.g. with a comment.

Reply:
Good observation

Reply:
https://github.com/PX4/Matrix/pull/126